app/components/learningmaterial-search.js

```diff
@@ -1,99 +1,101 @@
-        this.setProperties({
-          results: [],
-          showMoreInputPrompt: true,
-        });
-
-        return;
-      }
-
+      this.set('searchReturned', false);
       this.set('searching', true);
-
-      debounce(this, function() {
-        this.send('search', searchTerms);
-      }, 300);
+      this.set('query', query);
+      var currentMaterials = this.get('currentMaterials');
+      this.get('store').query('learningMaterial', {
+        q: query,
+        limit: this.get('searchResultsPerPage') + 1,
+        'order_by[title]': 'ASC',
+      }).then(function(lms){
+        let results = lms.map(function(lm){
+          return ProxiedMaterials.create({
+            content: lm,
+            isActive: ! currentMaterials.contains(lm)
+          });
+        });
+        self.set('searchReturned', true);
+        self.set('searching', false);
+        self.set('searchPage', 1);
+        self.set('hasMoreSearchResults', (results.length > self.get('searchResultsPerPage')));
+        if (self.get('hasMoreSearchResults')) {
+          results.pop();
+        }
+        self.set('searchResults', results);
+      });
     },
-
-    search(searchTerms) {
-      const store = this.get('store');
-
-      this.set('searching', true);
-      store.query('learningMaterial', { q: searchTerms }).then((learningMaterials) => {
-        let results = learningMaterials.filter((learningMaterial) => {
-          return !this.get('currentMaterials').contains(learningMaterial);
+    searchMore: function() {
+      var self = this;
+      var currentMaterials = this.get('currentMaterials');
+      var query = this.get('query');
+      this.get('store').query('learningMaterial', {
+        q: query,
+        limit: this.get('searchResultsPerPage') + 1,
+        offset: this.get('searchPage') * this.get('searchResultsPerPage'),
+        'order_by[title]': 'ASC',
+      }).then(function(lms){
+        let results = lms.map(function(lm){
+          return ProxiedMaterials.create({
+            content: lm,
+            isActive: ! currentMaterials.contains(lm)
+          });
         });
-
-        this.set('searching', false);
-
-        if (results.get('length') === 0) {
-          this.set('showNoResultsMessage', true);
+        self.set('searchPage', self.get('searchPage') + 1);
+        self.set('hasMoreSearchResults', (results.length > self.get('searchResultsPerPage')));
+        if (self.get('hasMoreSearchResults')) {
+          results.pop();
         }
-
-        this.set('results', results.sortBy('title'));
+        self.get('searchResults').pushObjects(results);
       });
     },
+    clear: function(){
+      this.set('searchResults', []);
+      this.set('searchReturned', false);
+      this.set('searching', false);
+      this.set('searchPage', 0);
+      this.set('hasMoreSearchResults', false);
+      this.set('query', '');
+    },
 
-    add(lm) {
-      this.sendAction('add', lm);
+    add(proxy) {
```

Comment:
Add should be a task which yields the result of sendAction('add') and queues up repeated clicks to ensure we don't try and add the same thing twice.